crates/oxc_query/src/vertex.rs

```diff
@@ -0,0 +1,319 @@
+            Vertex::PathPart(_) => "PathPart".into(),
+            Vertex::SearchParameter(_) => "SearchParameter",
+            Vertex::Span(_) => "Span",
+            Vertex::SpecificImport(_) => "SpecificImport",
+            Vertex::TypeAnnotation(_) => "TypeAnnotation",
+            Vertex::Type(_) => "Type",
+            Vertex::URL(_) => "URL",
+            Vertex::VariableDeclaration(_) => "VariableDeclaration",
+            Vertex::ReturnStatementAST(_) => "ReturnStatementAST",
+        }
+    }
+
+    pub fn span(&self) -> Span {
+        match &self {
+            Self::AssignmentType(data) => data.span(),
+            Self::ASTNode(data) => data.kind().span(),
+            Self::Class(data) => data.class.span,
+            Self::ClassMethod(data) => data.method.span,
+            Self::ClassProperty(data) => data.property.span,
+            Self::DefaultImport(data) => data.span,
+            Self::Expression(data) => data.span(),
+            Self::Import(data) => data.import.span,
+            Self::Interface(data) => data.interface.span,
+            Self::InterfaceExtend(data) => match **data {
+                InterfaceExtendVertex::Identifier(ident) => ident.span,
+                InterfaceExtendVertex::MemberExpression(membexpr) => (*membexpr).span(),
+            },
+            Self::JSXAttribute(data) => data.span,
+            Self::JSXElement(data) => data.element.span,
+            Self::JSXExpressionContainer(data) => data.span,
+            Self::JSXFragment(data) => data.span,
+            Self::JSXOpeningElement(data) => data.span,
+            Self::JSXSpreadAttribute(data) => data.span,
+            Self::JSXSpreadChild(data) => data.span,
+            Self::JSXText(data) => data.span,
+            Self::ObjectLiteral(data) => data.span,
+            Self::SpecificImport(data) => data.span,
+            Self::TypeAnnotation(data) => data.type_annotation.span,
+            Self::Type(data) => data.span(),
+            Self::URL(_) => unreachable!(),
+            Self::VariableDeclaration(data) => data.variable_declaration.span,
+            Self::ReturnStatementAST(data) => data.return_statement.span,
+            Self::File | Self::PathPart(_) | Self::SearchParameter(_) | Self::Span(_) => {
+                unreachable!()
+            }
+        }
+    }
+
+    pub fn ast_node_id(&self) -> AstNodeId {
+        match &self {
+            Vertex::ASTNode(data) => data.id(),
+            Vertex::Class(data) => data.ast_node.unwrap().id(),
+            Vertex::Import(data) => data.ast_node.unwrap().id(),
+            Vertex::Interface(data) => data.ast_node.unwrap().id(),
+            Vertex::JSXElement(data) => data.ast_node.unwrap().id(),
+            Vertex::TypeAnnotation(data) => data.ast_node.unwrap().id(),
+            Vertex::VariableDeclaration(data) => data.ast_node.unwrap().id(),
+            Vertex::ReturnStatementAST(data) => data.ast_node.unwrap().id(),
```

Comment:
consider `expect()` with a suitable message instead of `unwrap()`

Reply:
I ended up just making this function return an option which I then .expect() later, here https://github.com/web-infra-dev/oxc/pull/599/files#diff-eefcd299dc88179fdb05834ce810448eeb07394a9354e56e13eb57ad8e8e38aaR97